src/language-js/parse/postprocess/index.js

```diff
@@ -79,7 +95,9 @@ function postprocess(ast, options) {
       }
       // remove redundant TypeScript nodes
       case "TSParenthesizedType": {
-        node.typeAnnotation.range = [locStart(node), locEnd(node)];
+        if (!tsKeywordTypes.has(node.typeAnnotation.type)) {
+          node.typeAnnotation.range = [locStart(node), locEnd(node)];
```

Comment:
This is required for some formatting (See https://github.com/prettier/prettier/blob/main/tests/format/misc/typescript-babel-only/prettier-ignore-nested-unions.ts and https://github.com/prettier/prettier/blob/main/tests/format/misc/typescript-babel-only/prettier-ignore-parenthesized-type.ts). But `TSParenthesizedType` is no longer supported by `typescript-estree`. So we can remove this?

Reply:
If we remove this, will it output ugly?

Reply:
Maybe yes, in some formats in `babel-ts`

Reply:
I reconsidered. Let's keep this. It is possible to remove it in v3, but we shouldn't do it here.